homeassistant/components/broadlink/__init__.py

```diff
@@ -1 +1,86 @@
 """The broadlink component."""
+import asyncio
+from base64 import b64decode, b64encode
+import logging
+import re
+
+from datetime import timedelta
+import voluptuous as vol
+
+from homeassistant.const import CONF_HOST
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util.dt import utcnow
+
+from .const import DOMAIN, SERVICE_LEARN, SERVICE_SEND
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def ipv4_address(value):
+    """Validate an ipv4 address."""
+    regex = re.compile(r'^\d+\.\d+\.\d+\.\d+$')
+    if not regex.match(value):
+        raise vol.Invalid('Invalid Ipv4 address, expected a.b.c.d')
+    return value
+
+
+def data_packet(value):
+    """Decode a data packet given for broadlink."""
+    return b64decode(cv.string(value))
+
+
+SERVICE_SEND_SCHEMA = vol.Schema({
+    vol.Required(CONF_HOST): ipv4_address,
+    vol.Required('packet'): vol.All(cv.ensure_list, [data_packet])
+})
+
+SERVICE_LEARN_SCHEMA = vol.Schema({
+    vol.Required(CONF_HOST): ipv4_address,
+})
+
+
+def async_setup_service(hass, host, device):
```

Comment:
FYI: This is unconventional. For the long term we want to have a real component that is set up instead. For now I think it's ok like this.

Reply:
Yes, but that would need to wait until we move config to the component instead of the platform.